lib/modules/datasource/pypi/index.ts

```diff
@@ -182,10 +182,14 @@ export class PypiDatasource extends Datasource {
     // pep-0427 wheel packages
     //  {distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl.
     const wheelText = text.toLowerCase();
-    const wheelPrefix = packageName.replace(regEx(/[^\w\d.]+/g), '_') + '-';
```

Comment:
Next question: was the `.` in this regex correct in the first place? Should it have instead been `/.`?

I think we need to document better here, so that it's clearer to future maintainers. When I look at this code (and some of the old code) I'm not sure what/why it's meant to be doing.

Reply:
Answering that question as I also need this PR merged in order for some internal packages to be updated correctly by renovate.

Within a character class the only special characters are `^-]\` so a `.` only matches a `.`.
`\.` will still work and possibly be more understandable for most people, however the current regex does match the regex string in the PEP427 example exactly which I think is preferable: https://peps.python.org/pep-0427/#escaping-and-unicode

I do agree that an extra comment or two couldn't hurt. Particularly as this is trying to match both the older specification (PEP427) still being used by most packages as well as the current Python Binary Distribution Format specification which won't be terribly clear to people who aren't familiar with wheel naming conventions.